File: CS/WindowsFormsApplication1/XtraReport1.cs
using System.Drawing.Printing;
// ...

namespace WindowsFormsApplication1 {
    public partial class XtraReport1 : DevExpress.XtraReports.UI.XtraReport {
        public XtraReport1() {
            InitializeComponent();
        }

        private void lbShowHide_BeforePrint(object sender, PrintEventArgs e) {
            lbShowHide.Text = DetailReport.DrillDownExpanded ? "Hide Details" : "Show Details";
        }

    }
}

File: VB/WindowsFormsApplication1/XtraReport1.vb
Imports System.Drawing.Printing
' ...

Namespace WindowsFormsApplication1
    Partial Public Class XtraReport1
        Inherits DevExpress.XtraReports.UI.XtraReport

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub lbShowHide_BeforePrint(ByVal sender As Object, ByVal e As PrintEventArgs) Handles lbShowHide.BeforePrint
            lbShowHide.Text = If(DetailReport.DrillDownExpanded, "Hide Details", "Show Details")
        End Sub

    End Class
End Namespace
